### ISSUE TITLE
[Bug]: Fixture allows for a registered domain with "Investigating" status

### ISSUE THREAD

**Issue by igorkorenfeld** [cisagov/getgov#701]
### Current Behavior

When doing work on a domain management page I noticed that the domain listed under "Registered domain" had a status of of "Investigating." This shouldn't happen because a domain request that has been approve should have an "Approved" status, and while it is being investigated, it shouldn't appear in the registered domains area (and there should not be a corresponding domain created yet).

![image](https://github.com/cisagov/getgov/assets/52677065/f55d6396-2462-4e63-8e16-0dcb39d00632)

### Expected Behavior

Expected the registered domain to have an "Accepted" status

### Steps to Reproduce

This appears in my local environment. I am not sure how exactly to replicate it. 

### Environment (optional)

Environment: Local

may relate to #796

**Comment by kitsushadow** [comment]
"Investigating" will be referenced as "In Review"

**Comment by Katherine-Osos** [comment]
This may get fixed when #796 is completed.

**Comment by rachidatecs** [comment]
Did some investigation: The issue is that in fixtures, a IN REVIEW application is set to run approve() which creates a domain as a side effect. The domain is getting created, the application status is set to approved, but on local only that application status does not persist.

We assumed it's possible that #756 may be playing a role and overwriting that status somehow, but from logging this does not seem to be the case.

Tried this in signals, the logs show approved for instance.status:

```
@receiver(post_transition, sender=DomainApplication)
def save_new_workflow_state(sender, instance, name, source, target, **kwargs):
    """Save the new workflow state following successful django_fsm transition."""
    logger.debug(f'POST TRANSITION source {source}')
    logger.debug(f'POST TRANSITION target {target}')
    logger.debug(f'POST TRANSITION instance.status {instance.status}')
    if source != target:
        logger.debug('SOURCE IS DIFFERENT THAN TARGET')
        instance.save() 
```

Also tried blocking new application creation attempts in fixtures when a related domain name exists, no cigar either:

```
class DomainApplicationFixture:
...
@classmethod
    def load(cls):
        """Creates domain applications for each user in the database."""
        logger.info("Going to load %s domain applications" % len(cls.DA))
        try:
            users = list(User.objects.all())  # force evaluation to catch db errors
        except Exception as e:
            logger.warning(e)
            return

        for user in users:
            logger.debug("Loading domain applications for %s" % user)
            for app in cls.DA:
                try:
                    
                    if DomainApplication.objects.filter(creator=user, status=DomainApplication.IN_REVIEW).exists():
                        logger.debug('DomainApplication with status IN REVIEW exists')
                        application = DomainApplication.objects.filter(
                            creator=user, status=DomainApplication.IN_REVIEW
                        ).last()
                        try:
                            linked_domain = Domain.objects.filter(name=application.requested_domain)
                            if linked_domain:
                                logger.debug('DA An associated domain exists, passing')
                                pass
                            else:
                                logger.debug(' DA No associated domains, creating DA')
                                da, _ = DomainApplication.objects.get_or_create(
                                    creator=user,
                                    organization_name=app["organization_name"],
                                )
                                cls._set_non_foreign_key_fields(da, app)
                                cls._set_foreign_key_fields(da, app, user)
                                da.save()
                                cls._set_many_to_many_relations(da, app)
                        except Exception as e:
                            logger.warning(e)
                    else:
                        logger.debug('DomainApplication with status IN REVIEW does not exist, creating DA')
                        da, _ = DomainApplication.objects.get_or_create(
                            creator=user,
                            organization_name=app["organization_name"],
                        )
                        cls._set_non_foreign_key_fields(da, app)
                        cls._set_foreign_key_fields(da, app, user)
                        da.save()
                        cls._set_many_to_many_relations(da, app)
                except Exception as e:
                    logger.warning(e)


class DomainFixture(DomainApplicationFixture):

    """Create one domain and permissions on it for each user."""

    @classmethod
    def load(cls):
        try:
            users = list(User.objects.all())  # force evaluation to catch db errors
        except Exception as e:
            logger.warning(e)
            return

        for user in users:
            # approve one of each users in review status domains
            application = DomainApplication.objects.filter(
                creator=user, status=DomainApplication.IN_REVIEW
            ).last()
            logger.debug(f"Approving {application} for {user}")
            logger.debug(f"Before APPROVE {application.status}")
            linked_domain = Domain.objects.filter(name=application.requested_domain)
            if linked_domain:
                logger.debug('Approving - An associated domain exists')

            else:
                logger.debug('Approving - No associated domains')
                application.approve()
                application.save()
                logger.debug(f"After APPROVE {application.status}")
```

**Comment by rachidatecs** [comment]
This got fixed with #756
